### ISSUE TITLE
Pal/Linux and Pal/Linux-SGX should share code of db_*.c

### ISSUE THREAD

**Issue by yamahata** [oscarlab/graphene#443]
db_*.c of Pal/Linux and Pal/Linux_SGX are mostly same and the difference is only host syscall or ocall.
Because they are maintained differently, there are several bugs in only one of them with other fixed.
this is very bad situation. we should consolidate them into one to reduce maintenance cost and to increase test coverage.

**Comment by donporter** [comment]
I agree with this.  It would be good to hoist more similar code that has a common function into a shared code base.

**Comment by mkow** [comment]
+1, this is definitely a huge problem with our current codebase.

**Comment by dimakuv** [comment]
Just to add to this, I recently submitted a PR #1091 which re-writes the threading of SGX PAL to closely resemble the threading of Linux PAL. So the PR on sharing of code should be done after #1091 is merged.

**Comment by dimakuv** [comment]
Duplicate of #895. Let's close this issue and track all progress in #895.
